### PR TITLE
adding gsl and bioconductor-dirichletmultinomial

### PR DESCRIPTION
**Pull request by jerowe**
- [ ] I have read the guidelines above.
- [x] This PR adds a new recipe.
- [ ] This PR updates an existing recipe.
- [ ] This PR does something else (explain below).

https://www.gnu.org/software/gsl/
http://bioconductor.org/packages/release/bioc/html/DirichletMultinomial.html

### CODE REVIEW

**Comment by bgruening** [comment]
You can rerun travis if you are a commiter, you are isn't it?

**Comment by bgruening** [comment]
Otherwise a general trick for travis is to close and reopen a PR :)

File: recipes/gsl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package:
```
Comment (bgruening) [inline recipes/gsl/meta.yaml]: Do you need gsl 2.2? gsl is available in conda-forge and I think the correct way would be to upgrade this version.

File: recipes/bioconductor-dirichletmultinomial/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - lib/R/lib/
+    - lib/
+requirements:
+  build:
+    - bioconductor-biocgenerics
+    - bioconductor-iranges
+    - bioconductor-s4vectors
+    - r
+  run:
+    - bioconductor-biocgenerics
```
Comment (bgruening) [inline recipes/bioconductor-dirichletmultinomial/meta.yaml]: This one is compiling, isn't it? So you need to add gcc/llvm?

**Comment by jerowe** [comment]
I didn't realize gsl was in conda-forge. I removed it from the PR.

I suppose I could be smart about restarting travis, but I always just push a new commit. ;-)

**Comment by jerowe** [comment]
@bgruening , any idea why this is failing?

**Comment by daler** [comment]
@jerowe see https://github.com/bioconda/bioconda-recipes/pull/2729#issuecomment-257187874, @molpopgen is running into GSL issues as well . Maybe `{{CONDA_GSL}}` would work?

**Comment by molpopgen** [comment]
Also see #2590.  I think the Linux builds are not pulling from conda forge.  GSL diagnoses the problem because conda version is broken, causing failures seen here.

**Comment by jerowe** [comment]
@daler and @molpopgen , thanks for the input. I am trying now with CONDA_GSL.

**Comment by jerowe** [comment]
FYI, when I build on my local docker image I get

gsl:                       1.16-1           r  

So it looks like it is being pulled from r.

**Comment by jerowe** [comment]
It works with {{CONDA_GSL}}

Thanks!

**Comment by bgruening** [comment]
Awesome!

**Pull request merged**